drgn_tools/lsmod.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import argparse
+from typing import Iterable
+from typing import List
+
+from _drgn import FaultError
+from drgn import cast
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.list import list_for_each_entry
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.module import KernelModule
+from drgn_tools.table import print_table
+
+
+def for_each_module_use(
+    prog: Program, source_list_addr: Object
+) -> Iterable[Object]:
+    """
+    Provide the list of ``struct module_use`` as an iterable object
+
+    :param source_list_addr: ``struct module_use.source_list `` Object.
+    :returns: A list of ``struct module.source_list`` as an iterable object
+    """
+    return list_for_each_entry(
+        "struct module_use", source_list_addr, "source_list"
+    )
+
+
+def print_module_parameters(prog: Program) -> None:
+    """
+    Provide the list of ``module parameters`` as an iterable object
+
+    :param prog: drgn program.
+    :returns: None
+    """
+
+    # Module details
+    for mod in KernelModule.all(prog):
+        print("\n")
+        print("MODULE NAME:".ljust(15), mod.name)
+        print("PARAM COUNT:", str(mod.obj.num_kp.value_()))
+        print("ADDRESS    :", hex(mod.obj.num_kp.address_of_()))
+        table_value = []
+        ker_parm = hex(mod.obj.kp.value_())
+        table_value.append(["PARAMETER", "ADDRESS", "TYPE", "VALUE"])
+        integer_params = {
+            "byte": "unsigned char *",
+            "short": "short *",
+            "ushort": "unsigned short *",
+            "int": "int *",
+            "uint": "unsigned int *",
+            "long": "long *",
+            "ulong": "unsigned long *",
+        }
+        for i in range(0, mod.obj.num_kp):
+            parm_valf = ""
+            parm_val = ""
+            kp = mod.obj.kp[i]
+            try:
+                param_type = prog.symbol(kp.ops.get).name
+            except LookupError:
+                continue
```

Comment:
In the case of a LookupError, I think we can still print out the name, and indicate that we didn't know what to do with it. EG:

```python
table_value.append([kp.name.string_().decode("utf-8"), kern_parm, "UNKNOWN", "UNKNOWN"])
continue
```

This way, module parameters are not skipped over just because we had trouble with them.